network/p2p/libp2pConnector.go

```diff
@@ -17,13 +17,14 @@ import (
 )
 
 // libp2pConnector is a libp2p based Connector implementation to connect and disconnect from peers
-type libp2pConnector struct {
+type Libp2pConnector struct {
 	backoffConnector *discovery.BackoffConnector
 	host             host.Host
 	log              zerolog.Logger
+	pruneConnections bool
 }
 
-var _ Connector = &libp2pConnector{}
+var _ Connector = &Libp2pConnector{}
```

Comment:
@vishalchangrani what is the purpose of this?

Reply:
this is a compile time check to make sure that Libp2pConnector implements the Connector interface ()